src/pages/blog/the-search-for-performance.md

```diff
@@ -0,0 +1,64 @@
+
+I'll present a mild warning at this moment, all numbers below are from a virtual machine on my personal desktop. I may, or may not had a StarShip launch stream running in the background on this computer, but I was able to reproduce these numbers under a few different conditions, even with the headache of Thermal Throttling was a thing.
+
+### Database Indexes, Result sets, and SQLAlchemy ###
+
+And so, I started looking at adding indexes. This was surprisingly easy for me as someone who has spent substantial time working on databases. The headache was creating some mock data to exercise this, and then measuring it.
+
+Then I had to remember the way SQLAlchemy worked when you use an Object Relationship Model. Long story short, it downloads the entire object, and then assembles the object. So indexes only help so much. They *help* for query concurrency and raw result set generation. And then I remembered, exactly why they say to never run "select *" on your SQL queries, because all of the columns are collected for the result set.
+
+The answer was to restrict the query down to specific columns. Turns out, that is not entirely honored in Ironic's database access code. This shall be fixed.
+
+### Red herrings of Objects ###
+
+When consulting the SQLAlchemy documentation, it is easy to to jump down the path of thinking that all of the object generations are part of the latency between getting the result set, downloading the result set, and then converting the result set to objects, which are then transformed and supplied to the API consumer.
+
+Exploring this, and after numerous print statements into the code, it became clear the policy enforcement and sanitization code *uses* a substantial portion of the time to generate and return the result sets.
+
+And then I started beating myself up, since this was code I recently modified. I created part of this issue, or at least compounded it further.
+
+So I have a path forward, to just keep things simple and only do the needful, not extra work. It turned out taking this approach makes things even *faster*. We tend to error on the side of caution. To always expect to do certian tasks, but rarely stop to check if we *need* to take an extra step.
+
+## End results ##
+
+In the end, It appears we're able to squeeze out quite a bit more performance out of the API through a series of changes and improvements. Some of these these changes may not be feasible. Some may take a little more work. Most *should* be able to be backported to older versions, but that remains to be seen. In the mean time, I'll leave you with a graph.... including a little more fine tuning and addressing the afternoon heat causing some thermal throttling.
+
+<img src="../../img/blog-performance.png" alt="Graph of database and api nodes returned per second via the Ironic API">
```

Comment:
@jimmytipit Hey, it looks like it is injecting static in to the path no matter what I do. Any hints?

Reply:
Try this format:

src="/img/blog-performance.png"

Reply:
Tried it on the first rev, which is why it was confusing. :\

Reply:
This shouldn't make a difference, but it might b/c of the way netlify ultimately renders the site. Try this...


![Graph of database and api nodes returned per second via the Ironic API](/img/blog-performance.png)

Reply:
@jimmytipit That worked like a charm!